In [1]:
import numpy as np
import pandas as pd

import warnings
warnings.filterwarnings('ignore')

In [2]:
df= pd.read_csv(r'employee_data.csv')
df.head()

AGENT_ID      AGENT_NAME DATE_OF_JOINING          ADDRESS_LINE1  \
0  AGENT00001       Ray Johns      05-06-1993       1402 Maggies Way   
1  AGENT00002   Angelo Borjon      27-12-2005         414 Tanya Pass   
2  AGENT00003  Candy Spellman      02-09-2003    606 National Street   
3  AGENT00004      Mary Smith      23-09-2004  235 Hugh Thomas Drive   
4  AGENT00005    Mildred Diaz      21-06-2011  3426 Broadview Street   

  ADDRESS_LINE2              CITY STATE  POSTAL_CODE  EMP_ROUTING_NUMBER  \
0           NaN  Waterbury Center    VT         5677            34584958   
1           NaN       Panama City    FL        32404           107363763   
2          #306      Fayetteville    AR        72701            81744097   
3           NaN       Panama City    FL        32404            67563771   
4           NaN        Montgomery    AL        36110           114951317   

      EMP_ACCT_NUMBER  
0  HKUN51252328472585  
1  OPIS19290040088204  
2  YSCJ67489688482590  
3  ZANG21285355574581  
4  DZFS82244494451134

In [3]:
df.shape

(1200, 10)

In [4]:
df1= pd.read_csv(r'insurance_data.csv')
df1.head()

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
0  2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468    2015-06-23   
1  2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594    2018-04-21   
2  2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969    2019-10-03   
3  2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292    2016-11-29   
4  2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204    2011-12-26   

      LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  ...  \
0  2020-05-16  2020-05-21         Health          157.13          9000  ...   
1  2020-05-13  2020-05-18       Property          141.71         26000  ...   
2  2020-05-21  2020-05-26       Property          157.24         13000  ...   
3  2020-05-14  2020-05-19         Health          172.87         16000  ...   
4  2020-05-17  2020-05-22         Travel           88.53          3000  ...   

  CLAIM_STATUS INCIDENT_SEVERITY AUTHORITY_CONTACTED ANY_INJURY  \
0            A        Major Loss              Police          0   
1            A        Total Loss           Ambulance          1   
2            A        Total Loss              Police          0   
3            A        Minor Loss           Ambulance          0   
4            A        Major Loss              Police          0   

  POLICE_REPORT_AVAILABLE  INCIDENT_STATE   INCIDENT_CITY  \
0                       1              GA        Savannah   
1                       0              AL      Montgomery   
2                       1              CO  Grand Junction   
3                       0              GA        Savannah   
4                       1              TN       Nashville   

  INCIDENT_HOUR_OF_THE_DAY    AGENT_ID  VENDOR_ID  
0                        4  AGENT00413  VNDR00556  
1                        0  AGENT00769  VNDR00592  
2                       19  AGENT00883  VNDR00031  
3                       12  AGENT00278  VNDR00075  
4                       18  AGENT00636  VNDR00472  

[5 rows x 38 columns]

In [5]:
df1.shape

(10000, 38)

In [6]:
df2= pd.read_csv(r'vendor_data.csv')
df2.head()

VENDOR_ID              VENDOR_NAME                   ADDRESS_LINE1  \
0  VNDR00001  King, Proctor and Jones        2027 North Shannon Drive   
1  VNDR00002               Garcia Ltd          5701 East Shirley Lane   
2  VNDR00003               Cherry LLC            1217 Cottondale Road   
3  VNDR00004              Mays-Benson  227 West Montgomery Cross Road   
4  VNDR00005               Wilson PLC            23 North Hill Street   

  ADDRESS_LINE2          CITY STATE  POSTAL_CODE  
0            #5  Fayetteville    AR        72703  
1           NaN    Montgomery    AL        36117  
2           NaN    Montgomery    AL        36109  
3          #736      Savannah    GA        31406  
4           NaN     Nashville    TN        37210

In [7]:
df2.shape

(600, 7)

In [8]:
#task1

Merge the 3 dataset and create 1 view of data.
You can merge insurance_data.csv and employee_data.csv on
AGENT_ID You can merge insurance_data.csv and vendor_data.csv on
VENDOR_ID Note: Use left Outer join as not all claims require Vendor

In [9]:
df3= pd.merge(df, df1, on= 'AGENT_ID', how= 'left', suffixes= ('_CUSTOMER','_AGENT'))
data= pd.merge(df3, df2, on='VENDOR_ID',how='left')

In [10]:
data

AGENT_ID         AGENT_NAME DATE_OF_JOINING ADDRESS_LINE1_CUSTOMER  \
0     AGENT00001          Ray Johns      05-06-1993       1402 Maggies Way   
1     AGENT00001          Ray Johns      05-06-1993       1402 Maggies Way   
2     AGENT00001          Ray Johns      05-06-1993       1402 Maggies Way   
3     AGENT00001          Ray Johns      05-06-1993       1402 Maggies Way   
4     AGENT00001          Ray Johns      05-06-1993       1402 Maggies Way   
...          ...                ...             ...                    ...   
9995  AGENT01200  Virginia Stachura      25-05-1999        3579 2nd Avenue   
9996  AGENT01200  Virginia Stachura      25-05-1999        3579 2nd Avenue   
9997  AGENT01200  Virginia Stachura      25-05-1999        3579 2nd Avenue   
9998  AGENT01200  Virginia Stachura      25-05-1999        3579 2nd Avenue   
9999  AGENT01200  Virginia Stachura      25-05-1999        3579 2nd Avenue   

     ADDRESS_LINE2_CUSTOMER     CITY_CUSTOMER STATE_CUSTOMER  \
0                       NaN  Waterbury Center             VT   
1                       NaN  Waterbury Center             VT   
2                       NaN  Waterbury Center             VT   
3                       NaN  Waterbury Center             VT   
4                       NaN  Waterbury Center             VT   
...                     ...               ...            ...   
9995                    NaN         Edgewater             MD   
9996                    NaN         Edgewater             MD   
9997                    NaN         Edgewater             MD   
9998                    NaN         Edgewater             MD   
9999                    NaN         Edgewater             MD   

      POSTAL_CODE_CUSTOMER  EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  ...  \
0                     5677            34584958  HKUN51252328472585  ...   
1                     5677            34584958  HKUN51252328472585  ...   
2                     5677            34584958  HKUN51252328472585  ...   
3                     5677            34584958  HKUN51252328472585  ...   
4                     5677            34584958  HKUN51252328472585  ...   
...                    ...                 ...                 ...  ...   
9995                 21037            12226940  ALYY24945545778511  ...   
9996                 21037            12226940  ALYY24945545778511  ...   
9997                 21037            12226940  ALYY24945545778511  ...   
9998                 21037            12226940  ALYY24945545778511  ...   
9999                 21037            12226940  ALYY24945545778511  ...   

     INCIDENT_STATE  INCIDENT_CITY INCIDENT_HOUR_OF_THE_DAY  VENDOR_ID  \
0                FL     Lynn Haven                        4  VNDR00394   
1                MD    Glen Burnie                        5  VNDR00222   
2                MD         Arnold                        6  VNDR00440   
3                AR   Fayetteville                       18  VNDR00267   
4                OK  Oklahoma City                       18        NaN   
...             ...            ...                      ...        ...   
9995             AL     Montgomery                        9  VNDR00240   
9996             FL    Panama City                       13  VNDR00412   
9997             TN      Nashville                       19        NaN   
9998             OK   Midwest City                       18        NaN   
9999             OK  Oklahoma City                       19  VNDR00251   

                       VENDOR_NAME              ADDRESS_LINE1 ADDRESS_LINE2  \
0                         Hart PLC         1108 Slater Street           NaN   
1                   Webb-Henderson      2209 East Manor Drive           NaN   
2     Hawkins, Hansen and Gonzalez  188 Saint Matthews Avenue            #1   
3                      Mercado Inc     6401 Green Meadow Lane           NaN   
4                              NaN                        NaN           NaN   
...                            ...    

In [11]:
#task2

Business Leader wants to find Top 3 Insurance Type where we are getting
most insurance claims?

In [12]:
df1.INSURANCE_TYPE.value_counts()[:3]

Property    1692
Mobile      1692
Health      1690
Name: INSURANCE_TYPE, dtype: int64

In [13]:
#top 3 insurance claims are property with 1692 claims, mobile with 1692 claims and health with 1690 claims

In [14]:
#task3

Business Leader wants to find Top 5 States where we are getting most insurance claims
for customer belonging to HIGH(H) risk segment?

In [15]:
df1.columns

Index(['TXN_DATE_TIME', 'TRANSACTION_ID', 'CUSTOMER_ID', 'POLICY_NUMBER',
       'POLICY_EFF_DT', 'LOSS_DT', 'REPORT_DT', 'INSURANCE_TYPE',
       'PREMIUM_AMOUNT', 'CLAIM_AMOUNT', 'CUSTOMER_NAME', 'ADDRESS_LINE1',
       'ADDRESS_LINE2', 'CITY', 'STATE', 'POSTAL_CODE', 'SSN',
       'MARITAL_STATUS', 'AGE', 'TENURE', 'EMPLOYMENT_STATUS',
       'NO_OF_FAMILY_MEMBERS', 'RISK_SEGMENTATION', 'HOUSE_TYPE',
       'SOCIAL_CLASS', 'ROUTING_NUMBER', 'ACCT_NUMBER',
       'CUSTOMER_EDUCATION_LEVEL', 'CLAIM_STATUS', 'INCIDENT_SEVERITY',
       'AUTHORITY_CONTACTED', 'ANY_INJURY', 'POLICE_REPORT_AVAILABLE',
       'INCIDENT_STATE', 'INCIDENT_CITY', 'INCIDENT_HOUR_OF_THE_DAY',
       'AGENT_ID', 'VENDOR_ID'],
      dtype='object')

In [16]:
df1[df1['RISK_SEGMENTATION']=='H'].STATE.value_counts()[:5]

CA    148
AZ    105
FL    104
TN    100
AR     97
Name: STATE, dtype: int64

These are the 5 states with highest risk segmentation claims

CA

AZ

FL

TN

AR

In [17]:
#Task4

Business wants to create a new variable “COLOCATION” which will have following values IF
Customer State == Incident State == Agent Address State THEN 1 ELSE 0
Find the mean of this new column

In [18]:
data.columns

Index(['AGENT_ID', 'AGENT_NAME', 'DATE_OF_JOINING', 'ADDRESS_LINE1_CUSTOMER',
       'ADDRESS_LINE2_CUSTOMER', 'CITY_CUSTOMER', 'STATE_CUSTOMER',
       'POSTAL_CODE_CUSTOMER', 'EMP_ROUTING_NUMBER', 'EMP_ACCT_NUMBER',
       'TXN_DATE_TIME', 'TRANSACTION_ID', 'CUSTOMER_ID', 'POLICY_NUMBER',
       'POLICY_EFF_DT', 'LOSS_DT', 'REPORT_DT', 'INSURANCE_TYPE',
       'PREMIUM_AMOUNT', 'CLAIM_AMOUNT', 'CUSTOMER_NAME',
       'ADDRESS_LINE1_AGENT', 'ADDRESS_LINE2_AGENT', 'CITY_AGENT',
       'STATE_AGENT', 'POSTAL_CODE_AGENT', 'SSN', 'MARITAL_STATUS', 'AGE',
       'TENURE', 'EMPLOYMENT_STATUS', 'NO_OF_FAMILY_MEMBERS',
       'RISK_SEGMENTATION', 'HOUSE_TYPE', 'SOCIAL_CLASS', 'ROUTING_NUMBER',
       'ACCT_NUMBER', 'CUSTOMER_EDUCATION_LEVEL', 'CLAIM_STATUS',
       'INCIDENT_SEVERITY', 'AUTHORITY_CONTACTED', 'ANY_INJURY',
       'POLICE_REPORT_AVAILABLE', 'INCIDENT_STATE', 'INCIDENT_CITY',
       'INCIDENT_HOUR_OF_THE_DAY', 'VENDOR_ID', 'VENDOR_NAME', 'ADDRESS_LINE1',
       'ADDRESS_LINE2', 

In [19]:
data['COLOCATION'] = (data['STATE_CUSTOMER']==data['INCIDENT_STATE'])&(data['INCIDENT_STATE']==data['STATE_AGENT'])

In [20]:
#changing values from true to 1 and false to 0

In [21]:
data['COLOCATION']= data.COLOCATION.astype('int')
data.COLOCATION.mean()

0.0044

In [22]:
#the mean of colocation is 0.0044

In [23]:
#task5

Data entry error was detected in the data and you are required to correct it. If for any
claim transaction “AUTHORITY_CONTACTED” is NOT “Police” and POLICE_AVAILABLE == 1
Then Update “AUTHORITY_CONTACTED” to Police.

In [24]:
not_police= data[data['AUTHORITY_CONTACTED']!='Police']

In [25]:
row_update= not_police[not_police['POLICE_REPORT_AVAILABLE']==1].index

In [26]:
data.loc[row_update,'AUTHORITY_CONTACTED']='Police'

In [27]:
data[(data['POLICE_REPORT_AVAILABLE']==1)&(data['AUTHORITY_CONTACTED']!='Police')]

Empty DataFrame
Columns: [AGENT_ID, AGENT_NAME, DATE_OF_JOINING, ADDRESS_LINE1_CUSTOMER, ADDRESS_LINE2_CUSTOMER, CITY_CUSTOMER, STATE_CUSTOMER, POSTAL_CODE_CUSTOMER, EMP_ROUTING_NUMBER, EMP_ACCT_NUMBER, TXN_DATE_TIME, TRANSACTION_ID, CUSTOMER_ID, POLICY_NUMBER, POLICY_EFF_DT, LOSS_DT, REPORT_DT, INSURANCE_TYPE, PREMIUM_AMOUNT, CLAIM_AMOUNT, CUSTOMER_NAME, ADDRESS_LINE1_AGENT, ADDRESS_LINE2_AGENT, CITY_AGENT, STATE_AGENT, POSTAL_CODE_AGENT, SSN, MARITAL_STATUS, AGE, TENURE, EMPLOYMENT_STATUS, NO_OF_FAMILY_MEMBERS, RISK_SEGMENTATION, HOUSE_TYPE, SOCIAL_CLASS, ROUTING_NUMBER, ACCT_NUMBER, CUSTOMER_EDUCATION_LEVEL, CLAIM_STATUS, INCIDENT_SEVERITY, AUTHORITY_CONTACTED, ANY_INJURY, POLICE_REPORT_AVAILABLE, INCIDENT_STATE, INCIDENT_CITY, INCIDENT_HOUR_OF_THE_DAY, VENDOR_ID, VENDOR_NAME, ADDRESS_LINE1, ADDRESS_LINE2, CITY, STATE, POSTAL_CODE, COLOCATION]
Index: []

[0 rows x 54 columns]

In [28]:
#task6

Business wants to check the Claim Amount for deviation for each transaction, they would
like you to calculate as follow
CLAIM_DEVIATION = AVG_CLAIM_AMOUNT_FOR_LAST_30DAYS (same insurance type)
/ CURRENT_CLAIM_AMOUNT
If the value < 0.5 THEN CLAIM_DEVIATION = 1 ELSE 0
If there is less than 30 days of transaction history THEN
-1 Note: LAST_30DAYS does not include current day

In [29]:
data.TXN_DATE_TIME.max()

'2021-06-30 00:00:00'

In [30]:
#assuming current date to be 2021-06-30
current_date= pd.to_datetime('2021-06-30')

In [31]:
data['TXN_DATE_TIME']= pd.to_datetime(data['TXN_DATE_TIME'])

In [32]:
last30 = data.loc[(data['TXN_DATE_TIME'] > '2021-05-30') & (data['TXN_DATE_TIME'] < '2021-06-30')]  

In [33]:
# Getting average claim of last 30 days insurance type wise and storing it in dictionary
average_claim = last30.groupby('INSURANCE_TYPE').mean()['CLAIM_AMOUNT'].to_dict()

In [34]:
# defining a function for calculating claim_deviation
def Claim_deviation(row):
    if (current_date - row['TXN_DATE_TIME'])<pd.Timedelta(days=30):
        return -1
    claim_deviation = average_claim[row['INSURANCE_TYPE']] / row['CLAIM_AMOUNT']
    if claim_deviation < 0.5:
        return 1
    else:
        return 0

In [35]:
# Applying the function to the dataframe
data['CLAIM_DEVIATION'] = data.apply(Claim_deviation,axis=1)

In [36]:
data.CLAIM_DEVIATION.value_counts()

 0    9216
-1     784
Name: CLAIM_DEVIATION, dtype: int64

There are 9216 claims with deviation less than 0.5

784 claims with transaction history of less than 30 days

In [37]:
#task7

Find All Agents who have worked on more than 2 types of Insurance Claims. Sort them by
Total Claim Amount Approved under them in descending order

In [38]:
#calculating insurance types worked on of each agent
count_type = data.groupby('AGENT_ID')['INSURANCE_TYPE'].nunique()

In [39]:
# Calculating weather agent has more than 2 insurance types
agent_insurance_types = count_type>2

In [40]:
#getting list of agents who have worked on more than 2 types of insurance types
multi_type_agents = agent_insurance_types[agent_insurance_types].index

In [41]:
top_agents = data.groupby(['AGENT_ID','AGENT_NAME']).sum().loc[multi_type_agents][['CLAIM_AMOUNT']].sort_values('CLAIM_AMOUNT',ascending=False)


In [42]:
top_agents

CLAIM_AMOUNT
AGENT_ID   AGENT_NAME                    
AGENT00807 Don Filkins             528800
AGENT00679 Clara Barnett           489000
AGENT00771 Roger Burns             422100
AGENT00125 Salvador Soriano        400400
AGENT00789 Alison Baron            392900
...                                   ...
AGENT00910 Vincent Ragan            15500
AGENT00689 Connie Hunt              13700
AGENT00732 Sylvia Tran              11300
AGENT00706 Laura Staggs             10800
AGENT00604 Thelma Salinas            5500

[1164 rows x 1 columns]

In [43]:
#task8

Mobile & Travel Insurance premium are discounted by 10%
Health and Property Insurance premium are increased by 7%
Life and Motor Insurance premium are marginally increased by 2%
What will be overall change in % of the Premium Amount Collected for all these Customer?

Mobile & Travel Insurance premium are discounted by 10%
Health and Property Insurance premium are increased by 7%
Life and Motor Insurance premium are marginally increased by 2%
What will be overall change in % of the Premium Amount Collected for all these Customer?

In [44]:
earlier_collected_premium= data['PREMIUM_AMOUNT'].sum()

In [45]:
#total collected premium by each type
premium_type= data.groupby('INSURANCE_TYPE').sum()['PREMIUM_AMOUNT'].to_dict()
premium_type

{'Health': 252455.73,
 'Life': 125621.22,
 'Mobile': 15191.42,
 'Motor': 165391.98,
 'Property': 202285.83,
 'Travel': 124139.77}

In [46]:
#mobile andtravel insurance discounted by 10%
premium_type['Mobile']*=0.9
premium_type['Travel']*=0.9

In [47]:
#health and propery premium is increased by 7%
premium_type['Health']*=1.07
premium_type['Property']*=1.07

In [48]:
#life and motor insurance increased by 2%
premium_type['Life']*=1.02
premium_type['Motor']*=1.02

In [49]:
#premium after discount and increase
revised_premium= sum(premium_type.values())

In [50]:
(revised_premium-earlier_collected_premium)*100/earlier_collected_premium

2.6798588543858517

In [51]:
earlier_collected_premium, revised_premium

(885085.95, 908805.0042000001)

Change in overall premium amount is 2.67

In [52]:
#task9

Business wants to give discount to customer who are loyal and under stress due to Covid
19. They have laid down an eligibility Criteria as follow
IF CUSTOMER_TENURE > 60 AND EMPLOYMENT_STATUS = “N”
AND NO_OF_FAMILY_MEMBERS >=4 THEN 1 ELSE 0
Create a new column “ELIGIBLE_FOR_DISCOUNT” and find it
mean.

In [53]:
# calculating status for discount with given conditions
discount_status=(data['TENURE']>60)&(data['EMPLOYMENT_STATUS']=='N')&(data['NO_OF_FAMILY_MEMBERS']>=4)

In [54]:
# Creating an column ELIGIBLE_FOR_DISCOUNT with above values
data['ELIGIBLE_FOR_DISCOUNT'] = discount_status.astype('int')

In [55]:
data['ELIGIBLE_FOR_DISCOUNT'].mean()

0.0299

The mean is 0.0299, which means that only 2.99% customers are eligible for loan

In [56]:
#task1o

Business wants to check Claim Velocity which is defined as follow
CLAIM_VELOCITY = NO_OF_CLAIMS_IN_LAST30DAYS (for the current insurance type)
/ NO_OF_CLAIMS_IN_LAST3DAYS (for the current insurance type)
Note: LAST30DAYS & LAST3DAYS does not include current
day

In [57]:
current_date

Timestamp('2021-06-30 00:00:00')

In [58]:
data['CLAIM_STATUS'].value_counts()

A    9497
D     503
Name: CLAIM_STATUS, dtype: int64

In [59]:
# we had defined last 30 days data in #task6, similarly defining for last 3 days
last_3_days = data.loc[(data['TXN_DATE_TIME'] > current_date-pd.Timedelta(days=4)) & (data['TXN_DATE_TIME'] < current_date)]

In [60]:
# Calculating total approved claims insurance type wise in last 30 days
claims_in_last_30_days = pd.DataFrame.from_dict(last30[last30['CLAIM_STATUS']=='A'].groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST30DAYS'])


In [61]:
# Calculating total approved claims insurance type wise in last 3 days
claims_in_last_3_days = pd.DataFrame.from_dict(last_3_days[last_3_days['CLAIM_STATUS']=='A'].groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST3DAYS'])


In [62]:
approved_claims = claims_in_last_30_days.join(claims_in_last_3_days)

In [63]:
approved_claims['CLAIM_VELOCITY'] = approved_claims['NO_OF_CLAIMS_IN_LAST30DAYS']/approved_claims['NO_OF_CLAIMS_IN_LAST3DAYS']
approved_claims

NO_OF_CLAIMS_IN_LAST30DAYS  NO_OF_CLAIMS_IN_LAST3DAYS  \
Health                           128                         14   
Life                             128                         12   
Mobile                           127                          9   
Motor                            117                         12   
Property                         125                         12   
Travel                           123                         14   

          CLAIM_VELOCITY  
Health          9.142857  
Life           10.666667  
Mobile         14.111111  
Motor           9.750000  
Property       10.416667  
Travel          8.785714

claim velocity is calculated for only claims approved, similarly ot is calculated for all claims below.

In [64]:
All_claims30D = pd.DataFrame.from_dict(last30.groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST30DAYS'])

In [65]:
All_claims3D = pd.DataFrame.from_dict(last_3_days.groupby('INSURANCE_TYPE').size().to_dict(),orient='index',columns=['NO_OF_CLAIMS_IN_LAST3DAYS'])

In [66]:
total_claims=All_claims30D.join(All_claims3D)

In [67]:
total_claims['CLAIM_VELOCITY'] = total_claims['NO_OF_CLAIMS_IN_LAST30DAYS']/total_claims['NO_OF_CLAIMS_IN_LAST3DAYS']
total_claims

NO_OF_CLAIMS_IN_LAST30DAYS  NO_OF_CLAIMS_IN_LAST3DAYS  \
Health                           136                         14   
Life                             140                         14   
Mobile                           134                         10   
Motor                            122                         13   
Property                         129                         13   
Travel                           125                         14   

          CLAIM_VELOCITY  
Health          9.714286  
Life           10.000000  
Mobile         13.400000  
Motor           9.384615  
Property        9.923077  
Travel          8.928571

In [68]:
#task11

Find all low performing agents i.e. employees who are in the bottom 5 percentile based
on Claims worked by them.

In [69]:
bottom_5percentile = data.groupby('AGENT_ID').sum().quantile(0.05)['CLAIM_AMOUNT']

In [70]:
Agent_grouped_data = data.groupby('AGENT_ID').sum()

In [71]:
# Low performing agent data
low_performing_agents = Agent_grouped_data[Agent_grouped_data['CLAIM_AMOUNT']<=bottom_5percentile][['CLAIM_AMOUNT']]
low_performing_agents

CLAIM_AMOUNT
AGENT_ID                
AGENT00006         22400
AGENT00014          7400
AGENT00016         27000
AGENT00085         25700
AGENT00088         27700
AGENT00091         30000
AGENT00098         28200
AGENT00134         22000
AGENT00149         28900
AGENT00156         13200
AGENT00172         22200
AGENT00185         16100
AGENT00186         27700
AGENT00188         28400
AGENT00197         18500
AGENT00252         20000
AGENT00263         11500
AGENT00275         16200
AGENT00279         28000
AGENT00300         12600
AGENT00310         28800
AGENT00316         22200
AGENT00351         21700
AGENT00425         17800
AGENT00442         30300
AGENT00536         20000
AGENT00562         21100
AGENT00571         24200
AGENT00585         16600
AGENT00590         22200
AGENT00604          5500
AGENT00621          5300
AGENT00628          6100
AGENT00672         21300
AGENT00689         13700
AGENT00706         10800
AGENT00721          9000
AGENT00732         11300
AGENT00773         15000
AGENT00780         11300
AGENT00790          6000
AGENT00804         20000
AGENT00805         22600
AGENT00830         23500
AGENT00841         17800
AGENT00856         24200
AGENT00894         27600
AGENT00910         15500
AGENT00969         28300
AGENT00985         30000
AGENT01021         24300
AGENT01037          5600
AGENT01052         28000
AGENT01077          4200
AGENT01107         24700
AGENT01128         28000
AGENT01131         27500
AGENT01151         28000
AGENT01154           800
AGENT01179         27300

In [72]:
#task12

Business wants to find all Suspicious Employees (Agents).
IF TOTAL CLAIM AMOUNT which meet below criteria is >= 15000 THEN AGENT is classified
as Suspicious ELSE Not
CLAIM_STATUS = Approved AND CUSTOMER_RISK_SEGMENTATION = High
AND INCIDENT_SEVERITY = “Major Loss”
If Suspicious, then 1 ELSE 0. Find mean of this column.

In [73]:
criteria = (data['CLAIM_STATUS']=='A')&(data['RISK_SEGMENTATION']=='H')&(data['INCIDENT_SEVERITY']=='Major Loss')

In [74]:
# transactions that follow criteria
criteria_met = data[criteria]

In [75]:
grouped_suspicion_data = criteria_met.groupby('AGENT_ID').sum()

In [76]:
# storing agents_id that are suspicious(met all the criteria)
suspicious_agents = list(grouped_suspicion_data[grouped_suspicion_data['CLAIM_AMOUNT']>=15000].index)

In [77]:
df

AGENT_ID         AGENT_NAME DATE_OF_JOINING  \
0     AGENT00001          Ray Johns      05-06-1993   
1     AGENT00002      Angelo Borjon      27-12-2005   
2     AGENT00003     Candy Spellman      02-09-2003   
3     AGENT00004         Mary Smith      23-09-2004   
4     AGENT00005       Mildred Diaz      21-06-2011   
...          ...                ...             ...   
1195  AGENT01196     Myrtle Mccatty      14-09-2005   
1196  AGENT01197        John Dorsey      04-12-1993   
1197  AGENT01198        Ruth Bigbee      05-09-1993   
1198  AGENT01199     Tommie Camacho      25-10-2004   
1199  AGENT01200  Virginia Stachura      25-05-1999   

                         ADDRESS_LINE1 ADDRESS_LINE2              CITY STATE  \
0                     1402 Maggies Way           NaN  Waterbury Center    VT   
1                       414 Tanya Pass           NaN       Panama City    FL   
2                  606 National Street          #306      Fayetteville    AR   
3                235 Hugh Thomas Drive           NaN       Panama City    FL   
4                3426 Broadview Street           NaN        Montgomery    AL   
...                                ...           ...               ...   ...   
1195  3553 Brandywine Street Northwest           NaN        Washington    DC   
1196              2315 Chestnut Street           NaN        Montgomery    AL   
1197            1983 Reidsville Street           NaN         Annapolis    MD   
1198                  5821 Taft Street           NaN            Arvada    CO   
1199                   3579 2nd Avenue           NaN         Edgewater    MD   

      POSTAL_CODE  EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  
0            5677            34584958  HKUN51252328472585  
1           32404           107363763  OPIS19290040088204  
2           72701            81744097  YSCJ67489688482590  
3           32404            67563771  ZANG21285355574581  
4           36110           114951317  DZFS82244494451134  
...           ...                 ...                 ...  
1195        20008            32137938  EWBR19907574832501  
1196        36107            70796687  KGEJ74846009867837  
1197        21401           124993912  RASV88064214164820  
1198        80004            19424059  UOFO11656245339211  
1199        21037            12226940  ALYY24945545778511  

[1200 rows x 10 columns]

In [78]:
df['SUSPICIOUS']=0

In [79]:
sus_emp_slice = df['AGENT_ID'].isin(suspicious_agents)

In [80]:
# Changing suspicious column value to 1 of suspicious employees
df.loc[sus_emp_slice,'SUSPICIOUS']=1

In [81]:
df['SUSPICIOUS'].mean()

0.12416666666666666

The mean of suspicious column is 0.124167 that means from all employees 12.42% are suspicious

In [1]:
pip install -U notebook-as-pdf

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.


In [3]:
pyppeteer-install


NameError: name 'pyppeteer' is not defined